robyn/router.py

```diff
@@ -124,11 +124,38 @@ def add_route(
         exception_handler: Optional[Callable],
         injected_dependencies: dict,
     ) -> Union[Callable, CoroutineType]:
+        def wrapped_handler(*args, **kwargs):
+            # In the execute functions the request is passed into *args
+            handler_params = signature(handler).parameters
+            # If no request arg, assume no other args are present
+            requests = list(filter(lambda a: isinstance(a, Request), args))
```

Comment:
How does this handle `INJECTED_DEPENDENCIES`?

Reply:
@kliwongan , one question what happens if there is no `requests, query_params` and just `query_params` ?

Reply:
It wouldn't work without the request object. If I did make it work without the request object it would require changes on the Rust side of things, so signatures would be checked before or during passing the args from the Rust side as the handler is being called

Reply:
As for router/global dependencies, I can make the change to handle them properly now; I can simply check for their existence first in the handler signature and then if they exist at all in the injected dependencies

Reply:
>It wouldn't work without the request object. If I did make it work without the request object it would require changes on the Rust side of things, so signatures would be checked before or during passing the args from the Rust side as the handler is being called

That kills the purpose of splitting the request params, unfortunately 😓

Reply:
The request handler is always(if there is one or more args) being passed from rust. No?

Reply:
> > It wouldn't work without the request object. If I did make it work without the request object it would require changes on the Rust side of things, so signatures would be checked before or during passing the args from the Rust side as the handler is being called
> 
> That kills the purpose of splitting the request params, unfortunately 😓

I can certainly make the necessary changes, it would simply move some, if not all of the changes over to the Rust side.

Reply:
> > > It wouldn't work without the request object. If I did make it work without the request object it would require changes on the Rust side of things, so signatures would be checked before or during passing the args from the Rust side as the handler is being called
> > 
> > 
> > That kills the purpose of splitting the request params, unfortunately 😓
> 
> I can certainly make the necessary changes, it would simply move some, if not all of the changes over to the Rust side.

Thank you! But before you proceed. How would you approach this task? Move the splitting logic before `function.call0(..)` ?

Reply:
I believe it would have to be placed there, yes, since if the request object isn't passed in to the handler, the only time to intercept it have to be prior to the call1

Reply:
@kliwongan 
Could you not just pass the request object to the handler from Rust and split it on the Python end?

Reply:
Oh yeah I suppose I can do that, thanks for pointing it out, I know exactly how do to it

Reply:
@sansyrox I have some code working and there's a bit of an ambiguity issue; for example, when there is a path whose handler function has only one unannotated argument, it's assumed that the single argument is meant for the Request object. So if we have a function that only has one argument meant to accept, say, only the request body, do we either:

1. Name it body (so we reserve specific keywords/names in the function argument section for splitting the request)
2.  OR name it whatever we want, but annotate it to identify what we want in that param (may need to create new enums or classes to represent each component of a Request for those components that don't have dedicated ones already)

Reply:
Hey @kliwongan 👋 

> do we either: Name it body (so we reserve specific keywords/names in the function argument section for splitting the request) OR name it whatever we want, but annotate it to identify what we want in that param (may need to create new enums or classes to represent each component of a Request for those components that don't have dedicated ones already)

We should do both. First check for annotations. If there are no annotations, we use the reserved keywords. We should prioritise annotations over reserved keywords.

Reply:
Sounds like a plan!